pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTableRestletResource.java

```diff
@@ -341,6 +344,54 @@ public SuccessResponse updateTableConfig(
 
     return new SuccessResponse("Table config updated for " + tableName);
   }
+  
+  /**
+   * Truncate table, delete all contents of table without removing table configuration and schema. 
+   * 
+   * @param tableName , name of the table to truncate
+   * @param tableTypeStr, type of the table i.e 'offline|realtime'
+   * @return response of truncate operation.
+   * @throws Exception
+   */
+  @POST
+  @Path("/tables/{tableName}/truncate")
+  @Produces(MediaType.APPLICATION_JSON)
+  @ApiOperation(value = "Truncate table")
+  public SuccessResponse truncateTable(
+      @ApiParam(value = "Name of the table to update", required = true) @PathParam("tableName") String tableName,
+      @ApiParam(value = "realtime|offline") @QueryParam("type") String tableTypeStr) throws Exception {
```

Comment:
We still have to define what truncate means for realtime. Typically, when you create a realtime table, it immediately starts consumption. So on truncate, after dropping all segments, we should again restart consumption based on configured offset. 
That will not happen with these changes. Consumption will be restarted only on the next run of the RealtimeSegmentValidationManager. Which I don't think is an acceptable flow.

Reply:
3 options
- this is ok as part of the version 
- we can invoke the validation manager at the end of the call
- we can add an separate call to pause/resume Kafka consumption

Reply:
Can we take the discussion to the Issue? I believe we need to agree on a spec first before we start publishing PRs.